v3/src/models/data/data-set.ts

```diff
@@ -279,6 +279,18 @@ export const DataSet = types.model("DataSet", {
     return collectionId === self.ungrouped.id ? self.ungrouped : getGroupedCollection(collectionId)
   }
 
+  function getGroupedCollectionByName(name: string): ICollectionModel | undefined {
+    return self.collections.find(collection => collection.name === name)
+  }
+
+  function getCollectionByName(name: string): ICollectionPropsModel | undefined {
+    if (!isAlive(self)) {
+      console.warn("DataSet.getCollectionByName called on a defunct DataSet")
+      return
```

Comment:
I didn't give this too much thought and just went with the path of least resistance--I mimicked `getGroupedCollection` and `getCollection` which are defined right above these, then used `getGroupedCollection` and `getGroupedCollectionByName` because they return `ICollectionModel` instead of `ICollectionPropsModel`.